Fixed pscid --include behaviour.

jamieyung opened the pull request:
Added an ADT (CLICommand) to represent the possible command types:
NamedScriptSpecificCommand ("pscid:" script in package.json),
NamedScriptCommand (script in package.json), and
PulpCommand (eg. "pulp build").

Only the PulpCommand data constructor has an associated array of
include paths; the reasoning is described in a comment
(Options.purs:93).

These commands are transformed into the appropriate string command via
a Show instance.

jamieyung committed:
Fixed pscid --include behaviour.

Added an ADT (CLICommand) to represent the possible command types:
NamedScriptSpecificCommand ("pscid:" script in package.json),
NamedScriptCommand (script in package.json), and
PulpCommand (eg. "pulp build").

Only the PulpCommand data constructor has an associated array of
include paths; the reasoning is described in a comment
(Options.purs:93).

These commands are transformed into the appropriate string command via
a Show instance.

kritzcreek commented on src/Pscid/Options.purs:
```diff
@@ -72,16 +73,50 @@ mkDefaultOptions =
+
+data CLICommand
+  = NamedScriptSpecificCommand String
+  | NamedScriptCommand String
```
Since the two NPM commands are treated the same everywhere I'd prefer only two variants here, maybe `PulpCommand` and `ScriptCommand`?

jamieyung replied:
Done.

kritzcreek commented on src/Pscid/Options.purs:
```diff
@@ -72,16 +73,50 @@ mkDefaultOptions =
+  | NamedScriptCommand String
+  | PulpCommand String (Array IncludePath)
+
+instance showCLICommand :: Show CLICommand where
```
Generally I only use `Show` instances for debugging. For a printer like this I prefer a simple function. How about `printCLICommand`?

jamieyung replied:
Done.

kritzcreek commented on src/Pscid/Options.purs:
```diff
@@ -121,18 +156,19 @@ buildOptions
              , sourceDirectories
              , censorCodes
-             , buildCommand: defaults.buildCommand
+             , buildCommand: buildCommand
```
If a variable with the same name as the field is in scope (like it is here in this PR) you can use a _record pun_ and avoid writing the same name twice.

`, buildCommand: buildCommand` -> `, buildCommand`

jamieyung replied:
Done.

kritzcreek commented on src/Pscid/Options.purs:
```diff
@@ -121,18 +156,19 @@ buildOptions
+             , buildCommand: buildCommand
              , outputDirectory
-             , testCommand: defaults.testCommand
+             , testCommand: testCommand
```
Another opportunity for a record pun, like above

jamieyung replied:
Done.

kritzcreek reviewed (changes requested):
Thank you so much, this looks great! Just a few stylistic comments and then this is good to go

kRITZCREEK commented:
Fixes #41

jamieyung committed:
Stylistic changes.

Collapsed NamedScriptSpecificCommand and NamedScriptCommand into just
ScriptCommand, since they are treated the same everywhere.

Replaced the Show instance for CLICommand with a function (to keep with
the convention of using Show instances for debugging).

Took a couple of opportunities to use record puns.

kRITZCREEK commented:
Thanks
